### PR TITLE
Add Nutanix Platform to Machine API Operator

### PR DESCRIPTION
**Pull request by thunderboltsid**
- Bump up API dependency to include Nutanix types
- Add Nutanix CredentialsRequest
- Add image for Nutanix Machine API Provider

### CODE REVIEW

File: install/0000_30_machine-api-operator_01_images.configmap.yaml [hunk 1]
```diff
@@ -23,5 +23,6 @@ data:
       "clusterAPIControllerIBMCloud": "registry.svc.ci.openshift.org/openshift:ibmcloud-machine-controllers",
       "clusterAPIControllerOvirt": "registry.svc.ci.openshift.org/openshift:ovirt-machine-controllers",
       "clusterAPIControllerPowerVS": "registry.svc.ci.openshift.org/openshift:powervs-machine-controllers",
-      "clusterAPIControllerVSphere": "registry.svc.ci.openshift.org/openshift:machine-api-operator"
+      "clusterAPIControllerVSphere": "registry.svc.ci.openshift.org/openshift:machine-api-operator",
+      "clusterAPIControllerNutanix": "public.ecr.aws/g7v6l0u0/openshift/mapi-provider-nutanix:latest"
```
Comment (thunderboltsid) [inline install/0000_30_machine-api-operator_01_images.configmap.yaml:27]: We probably should have a Openshift registry link for Nutanix MAPI provider too. I reckon we need to get https://github.com/nutanix-cloud-native/machine-api-provider-nutanix seeded over to the Openshift org.

**Comment by elmiko** [comment]
we need to make sure that the nutanix images are included in the payload before we merge this. i'm not sure on the status of that process, but i wanted to share some info from our internal discussions.

**Comment by thunderboltsid** [comment]
> 

@elmiko It's understandable that we wouldn't want to merge this before the image is included in the payload and the required tests pass. What I'm looking for at moment is whether we are touching the right bits in this PR that and not missing out on something obvious.

**Comment by elmiko** [comment]
@thunderboltsid sounds good, it looked good from my initial skim but i will make some time for a more thorough review.

**Review by elmiko — COMMENTED** [review]
/lgtm

/hold until images are in payload

File: install/0000_30_machine-api-operator_01_images.configmap.yaml [hunk 2]
```diff
@@ -23,5 +23,6 @@ data:
       "clusterAPIControllerIBMCloud": "quay.io/openshift/origin-ibmcloud-machine-controllers",
       "clusterAPIControllerOvirt": "quay.io/openshift/origin-ovirt-machine-controllers",
       "clusterAPIControllerPowerVS": "quay.io/openshift/origin-powervs-machine-controllers",
-      "clusterAPIControllerVSphere": "quay.io/openshift/origin-machine-api-operator"
+      "clusterAPIControllerVSphere": "quay.io/openshift/origin-machine-api-operator",
+      "clusterAPIControllerNutanix": "quay.io/openshift/origin-nutanix-machine-controllers"
```
Comment (JoelSpeed) [inline install/0000_30_machine-api-operator_01_images.configmap.yaml:27]: Could you add this at the correct place alphabetically please, I think the rest of this list is already alphabetical provider wise

**Comment by elmiko** [comment]
the images job has merged, i am removing the hold

https://github.com/openshift/release/pull/26964

/hold cancel

**Comment by JoelSpeed** [comment]
/hold

This image needs to be in payload before this PR can be merged else we will break builds across the OpenShift org.

Can we please link the ART requests for adding this to the payload so that they can be tracked alongside this PR?

File: install/0000_30_machine-api-operator_01_images.configmap.yaml (hunk 2 above)
Comment (thunderboltsid) [reply install/0000_30_machine-api-operator_01_images.configmap.yaml:27]: Sure

**Comment by JoelSpeed** [comment]
Code changes look good, will get the ART requests sorted this week to unblock this

**Comment by thunderboltsid** [comment]
> Code changes look good, will get the ART requests sorted this week to unblock this



> Code changes look good, will get the ART requests sorted this week to unblock this

Any updates on this?

**Comment by elmiko** [comment]
@thunderboltsid i am working to track down the details, in the meantime i have created a jira ticket to track the progress https://issues.redhat.com/browse/OCPCLOUD-1482

**Comment by elmiko** [comment]
i think the image issue has been worked out
/retest

**Comment by elmiko** [comment]
i guess we need a rebase here

**Comment by thunderboltsid** [comment]
> i guess we need a rebase here

Rebased against master; should resolve the conflict.

**Review by elmiko — COMMENTED** [review]
i think this is ready to go, we have the images created and the required tests are passing. @JoelSpeed would you mind taking another look at this on monday.
/lgtm

**Comment by JoelSpeed** [comment]
The Image is [still marked as WIP](https://github.com/openshift/ocp-build-data/blob/945ee9d12ae77f0cb6e69270b412de381eb14ae8/images/ose-nutanix-machine-controllers.yml#L1), I think if we merge this the image reference will break nightlies, can we please ascertain why this is still WIP please before we merge this

/hold

**Comment by JoelSpeed** [comment]
The WIP tag has now been removed, this is good to go

/approve
/hold cancel
/retest-required

**Pull request merged**